packages/backend/src/models/SpaceModel.ts

```diff
@@ -277,6 +358,39 @@ export class SpaceModel {
         await this.database(SpaceTableName)
             .update<UpdateSpace>(space)
```

Comment:
Should we clear the space share table here if the space changes from private to public? 

> Switching between these just switches between their defaults (e.g. if I have a restricted space where I've also invited Priyanka, then I switch it to "shared", then I switch it back to "restricted", it would restrict the access to just me, it wouldn't keep the old configurations and add Priyanka too)

Reply:
no, good catch

Reply:
![Screenshot from 2022-11-11 11-39-30](https://user-images.githubusercontent.com/1983672/201323711-a6b5e6d7-fbb7-48df-a1cd-50530b8330e0.png)
![Screenshot from 2022-11-11 11-40-55](https://user-images.githubusercontent.com/1983672/201323718-b70dd447-a024-4316-a114-926450953101.png)

Reply:
Not sure if we fixed the part where the space access needs to be reset when the space changes from private to public.

Reply:
oh right, I totally misunderstood the requirements (and your comment)... let me fix it.

Reply:
Done, I added `clear` access on isPrivate state change